semantic-core/src/Analysis/FlowInsensitive.hs

```diff
@@ -13,60 +13,62 @@ import           Control.Effect.Fresh
 import           Control.Effect.NonDet
 import           Control.Effect.Reader
 import           Control.Effect.State
-import qualified Data.Core as Core
 import qualified Data.Map as Map
 import           Data.Maybe (fromMaybe)
 import           Data.Monoid (Alt(..))
 import qualified Data.Set as Set
-import           Data.Term (Term)
 
-type Cache name a = Map.Map (Term (Core.Ann :+: Core.Core) name) (Set.Set a)
-type Heap name a = Map.Map name (Set.Set a)
+newtype Cache term a = Cache { unCache :: Map.Map term (Set.Set a) }
```

Comment:
Without the `newtype`, abstracting the cache over the term type made it impossible to distinguish from `Heap`.